### DISCUSSION TITLE
How to identify peak regions that fall in promoters and those that dont

### DISCUSSION THREAD

**Discussion by chrismahony** [stuart-lab/signac#993] [Q&A] [answered]
Hi

In my scATACseq assay I would like to identify peaks that fall within promoter regions (2kb up stream of TSS) and gene regions and peaks that dont fall into either (distal enhancers).

Ideally I would create a list of peaks that are or are not distal enhancers that I can use for futher clustering analysis and DE.

Thanks
Chris

**Comment by timoast** [comment, marked as answer]
You will first need to define the genomic ranges you're interested in for the gene promoters. If you're defining this as 2 kb upstream of the gene TSS, you can do something like:

```r
library(Signac)
library(GenomicRanges)

# get gene annotations
genes <- Annotations(atac_small)
promoters <- promoters(genes, upstream = 2000, downstream = 0)

# find distance to nearest promoter for each peak in the object
closest <- distanceToNearest(atac_small, promoters)
```

See [here](https://satijalab.org/signac/articles/data_structures.html#other-chromatinassay-methods-1) for some more examples of how to use granges functions with the Seurat object.

**Reply by jiangpuxuan** [reply]
Here is a little error:
```
genes <- Annotations(atac_small)
```
should be:
```
genes <- Annotation(atac_small)
```

**Comment by chrismahony** [comment]
Thanks!
